### PR TITLE
Travis config update, remove branch aliases

### PR DESCRIPTION
**Pull request by dnsl48**
https://github.com/silverstripe/cwp-recipe-kitchen-sink/issues/46#issuecomment-547734409

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
+    - php: 7.1
+      env: DB=MYSQL PHPCS_TEST=1 PHPUNIT_TEST=1
     - php: 7.2
-      env: DB=MYSQL RECIPE_VERSION=4.4.x-dev PHPUNIT_TEST=1
-    - php: 7.3
-      env: DB=MYSQL RECIPE_VERSION=4.5.x-dev PHPUNIT_TEST=1
+      env: DB=MYSQL PHPUNIT_TEST=1
     - php: 7.3
-      env: DB=MYSQL RECIPE_VERSION=4.4.x-dev MFA_VERSION=4.0.x-dev PHPUNIT_COVERAGE_TEST=1
+      env: DB=MYSQL PHPUNIT_COVERAGE_TEST=1
```
Comment (Cheddam) [inline .travis.yml]:
```suggestion
      env: DB=MYSQL MFA_VERSION=4.0.x-dev PHPUNIT_COVERAGE_TEST=1
```

**Review by Cheddam — CHANGES_REQUESTED** [review]
There's MFA support code in this module, so that module should be present in the code coverage build.

**Review by robbieaverill — DISMISSED** [review]
Sorry if I'm repeating myself, but where'd we end up with these points? You're proposing we stop testing multiple SilverStripe versions, that we remove branch aliases from master, and that we install from source instead of distributable packages in Travis. None of these are things I would happily accept as edits.

It appears (admittedly from the outside) as if you're trying to save some maintenance time by removing some (deliberate and) valuable things.

**Comment by Cheddam** [comment]
@robbieaverill 

### Testing multiple Silverstripe versions

The idea is to consolidate the tests running on the `master` branches, which reflect the next major going forward, and retain the existing version matrix on the 'current major' branches that @dnsl48 is rolling out. This way we unify the branching strategy with the core modules, and the master branches (which generally won't become relevant again until we approach SilverStripe 5) don't require updates after each minor release to keep them testing against a relevant base.

Ultimately the `master` branches will only run CI during merge-ups, and in those cases the current major branch will have done the hard work of testing compatibility with older core releases, so there's little value in running the full matrix again.

### Removing branch aliases

These currently require fiddling every time we trigger a new minor release. Packagist already manages producing branch aliases for numbered branches, so with the shift to having numbered current major branches across the board, this will be taken care of for us. For next majors, `dev-master` seems like a clear enough constraint that keeping the manual alias definitions isn't super beneficial.

### Installing from source instead of dist

I don't have a strong opinion here - the only differences I can see are that installing from source is more time-intensive (not great), and that it won't miss files that are excluded from the distributable (not really a problem anyway). @dnsl48 can you shed more light on the purpose behind this change?

**Comment by dnsl48** [comment]
> Installing from source instead of dist

This one is my overlook, sorry. That change was made to suit the original version of the PR with `vendor/bin/phpcs --standard=vendor/silverstripe/framework/phpcs.xml.dist`, but then we decided not to do so and when I was reverting the phpcs config usage, I forgot the `--prefer-dist` part of it.

This is my mistake, sorry. We're not gonna do `--prefer-source` for the major branches and potentially can revert this change in master branches too. I'm not sure it's too critical at this moment to perform this for all the modules right now, but we could do that when we have some better automation around patching multiple travis configs across the module base.

**Comment by dnsl48** [comment]
P.S. I'm gonna patch it for all not yet merged PRs

**Review by Cheddam — APPROVED** [review]

**Comment by Cheddam** [comment]
@robbieaverill Are you happy for these PRs to go forward based on our response?

**Comment by robbieaverill** [comment]
Yep, you guys do what you need to do - I'll just pop up occasionally and be annoying :)

**Pull request merged**